app/models/user.rb

```diff
@@ -9,4 +42,15 @@ class User < ApplicationRecord
 
   has_many :category_users
   has_many :categories, through: :category_users
+
+  before_save { self.email = email.downcase }
+  validates :name, presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
+  validates :sex, presence: true
+  validates :birthday, presence: true
+  validates :region, presence: true
+  validates :password_digest, presence: true
```

Comment:
```suggestion
  validates :password, presence: true, length: { minimum: 8 }
```

bycrptを使う場合は
password_digestはhash値を保存するカラムなのでvalidation
をかけることができません。(厳密にいうとできますが、hash値にvalidationをかけてもあまり意味がありません)
その代わりにpasswordというアクセサが生成されるので、
そのアクセサにvalidationをかける形となります。

今回の例はpassword 8文字以上の制限をつけた場合の例を書きました！